scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -2924,36 +2927,39 @@ u64 antistall_set(u64 dsq_id, u64 jiffies_now)
 			goto unlock;
 
 		#pragma unroll
-		for (pass = 0; pass < 2; ++pass) bpf_for(cpu, 0, nr_possible_cpus) {
-			const struct cpumask *cpumask;
-
-			if (!(cpumask = cast_mask(taskc->layered_mask)))
-				goto unlock;
-
-			/* for affinity violating tasks, target all allowed CPUs */
-			if (bpf_cpumask_empty(cpumask))
-				cpumask = p->cpus_ptr;
-
-			if (!bpf_cpumask_test_cpu(cpu, cpumask))
-				continue;
-
-			antistall_dsq = bpf_map_lookup_percpu_elem(&antistall_cpu_dsq, &zero_u32, cpu);
-			delay = bpf_map_lookup_percpu_elem(&antistall_cpu_max_delay, &zero_u32, cpu);
-
-			if (!antistall_dsq || !delay) {
-				scx_bpf_error("cant happen");
-				goto unlock;
-			}
-
-			if ((pass == 0 && *antistall_dsq == SCX_DSQ_INVALID) ||
-			    (pass != 0 && *delay < cur_delay)) {
-				trace("antistall set DSQ[%llu] SELECTED_CPU[%llu] DELAY[%llu]", dsq_id, cpu, cur_delay);
-				*delay = cur_delay;
-				*antistall_dsq = dsq_id;
-				goto unlock;
+		for (pass = 0; pass < 2; ++pass) {
+			for(antistall_multiplier = antistall_aggressiveness; antistall_multiplier > 0; --antistall_multiplier) {
+				bpf_for(cpu, 0, nr_possible_cpus) {
+					const struct cpumask *cpumask;
+					if (!(cpumask = cast_mask(taskc->layered_mask)))
+						goto unlock;
+
+					/* for affinity violating tasks, target all allowed CPUs */
+					if (bpf_cpumask_empty(cpumask))
+						cpumask = p->cpus_ptr;
+
+					if (!bpf_cpumask_test_cpu(cpu, cpumask))
+						continue;
+
+					antistall_dsq = bpf_map_lookup_percpu_elem(&antistall_cpu_dsq, &zero_u32, cpu);
+					delay = bpf_map_lookup_percpu_elem(&antistall_cpu_max_delay, &zero_u32, cpu);
+
+					if (!antistall_dsq || !delay) {
+						scx_bpf_error("cant happen");
+						goto unlock;
+					}
+
+					if ((pass == 0 && *antistall_dsq == SCX_DSQ_INVALID) ||
+			    		(pass != 0 && *delay < cur_delay)) {
+						trace("antistall set DSQ[%llu] SELECTED_CPU[%llu] DELAY[%llu]", dsq_id, cpu, cur_delay);
+						*delay = cur_delay;
+						*antistall_dsq = dsq_id;
+						if (antistall_aggressiveness <= 0)
```

Comment:
This is querying the constant which doesn't seem to be correct. I think the idea is that if the loop variable is on the last iteration terminate, but I'm still not sure that's right. We likely want to break out if the innermost `bpf_for` but still have another go, which I think means this unlock case needs to go.

Reply:
ahh yeah that should be multiplier